[IMP] mrp: Work order dependencies planning views (#17829)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/advanced_configuration/work_order_dependencies.rst

Reviewer: I know you didn't touch this section, but I had to manually enable the Start column in the Settings icon while working on a 19.2 runbot. Maybe worth mentioning?

The *work order dependencies* setting is not enabled by default. To enable it, navigate to
:menuselection:`Manufacturing app --> Configuration --> Settings`. Then, enable the
:guilabel:`Work Orders` setting if it is not already active.

After enabling the :guilabel:`Work Orders` setting, the :guilabel:`Custom Work Order Dependencies`

> setting appears below it. Enable :guilabel:`Custom Work Order Dependencies`, then click

Reviewer: Feels a little awkward having setting twice in one sentence. Maybe axe one or replace it with option or something?

Two work order planning views are available to display a visual representation of how work orders
are planned:

- :menuselection:`Manufacturing app --> Planning --> Gantt`
- :menuselection:`Manufacturing app --> Planning --> Kanban`

Both open different views of the *Work Orders Planning* page. Both the *Gantt* and *Kanban* views
show all work orders. The *Gantt* view shows a timeline of all the work orders scheduled for each
operation.

By default, the *Gantt* and *Kanban* views are grouped by *Work Center*. This grouping is best used
to view work orders grouped by the work center at which they take place. This view can help identify
overloaded machines or bottlenecks in production.

To view orders grouped by manufacturing order, click in the search bar and select

> :menuselection:`Group By --> Manufacturing Order`. This view is ideal for providing estimated

Reviewer: Worth calling out that the user may need to deselect  "Group By > Work Center"? It was the default in runbot and my experience in CRM tells me that the system changes what's displayed based on the order in which Group By things are selected.

delivery times or determining if an order will be completed on schedule.

In both views, if one work order is blocked by the completion of another, the blocked work order is

> shown as scheduled to start after the work order that blocks it.

Reviewer: This took a couple readthroughs for me to wrap my head around. What about a phrasing like:

`In both views, when one work order is blocked by the pending completion of another, the blocked order is shown as scheduled to begin once the pending order is complete.`

I don't love that, but what do you think?

Click the :guilabel:`Replan` button, then click :guilabel:`Save` to save the changes to the work
order. The work order is rescheduled after the previous work order in a way that maintains the work
order dependency.

The Kanban view
~~~~~~~~~~~~~~~

By default, the *Kanban* view is filtered to show only :guilabel:`Planned` work orders. To show work
orders that need to be planned, click in the search bar, then select :menuselection:`Filters --> To
Plan`.

In the *Kanban* view, blocked work orders are tagged with a yellow-orange circle in the bottom

> corner of the work order card.

Reviewer: It seems like you mention this earlier (or at least the yellow-orange circle.) Maybe it only needs to be mentioned once at the beginning of the doc? Idle thought, feel free to ignore as appropriate.

.. image:: work_order_dependencies/kanban-blocked-order.png
   :alt: A blocked work order with the yellow-orange circle.

> If a work order's planned start is in the past, and it has not started yet, its start date and time

Reviewer: ```suggestion
If a work order's planned start is in the past and it has not started yet, its start date and time
```

appear in red.

Plan unplanned manufacturing orders
***********************************

To plan confirmed manufacturing orders, click the :guilabel:`Plan Orders` button. The *Manufacturing
Orders* pop-up window opens. Select a manufacturing order to plan. The manufacturing order will be
planned so as not to disrupt the existing schedule.

Reschedule work orders
**********************

To reschedule work orders in the *Kanban* view, drag and drop the cards to their intended order,
then click the :guilabel:`Update Planning` button. To avoid introducing work order dependency
conflicts, the work order and any dependencies are replanned to maintain the correct order.

Work orders **can** be rescheduled without clicking the :guilabel:`Update Planning` button, but this

> may introduce conflicts, either via dependencies or scheduling work orders at the same time. To

Reviewer: ```suggestion
may introduce conflicts from dependencies or scheduling conflicts. To
```

Just to simplify it a bit.

determine if a work order is in conflict, open the work order card from the *Kanban* view. If a work
order is in conflict with another work order, a red :icon:`fa-exclamation-triangle`
:guilabel:`(exclamation triangle)` icon appears next to the :guilabel:`Start Date` fields.

> Resolve the conflict by clicking the :guilabel:`Update Planning` button on the Kanban view.

Reviewer: Feels like this ends a bit abruptly. There's no links or anything worth including to other docs?
